pkg/volume/glusterfs/glusterfs.go

```diff
@@ -974,6 +975,13 @@ func getClusterNodes(cli *gcli.Client, cluster string) (dynamicHostIps []string,
 	if len(dynamicHostIps) == 0 {
 		return nil, fmt.Errorf("no hosts found: %v", err)
 	}
+	// This should allow hostname or IP, See #4447.
+	for _, ipAddress := range dynamicHostIps {
```

Comment:
these IPs are obtained from 
```
ipaddr := dstrings.Join(nodeInfo.NodeAddRequest.Hostnames.Storage, "")
```

are those ips validated before they are added to the `Nodes`struct? do we need to validate them again if affirmative?

Reply:
commit updated, PTAL